indico/htdocs/js/indico/modules/surveys.js

```diff
@@ -53,6 +53,81 @@
         $('.js-add-question-dropdown').dropdown();
     }
 
+    global.setupSubmissionButtons = function setupSubmissionButtons() {
+        $('#export-submissions').on('click', function(evt) {
+            var $this = $(this);
+            var submissionIds = $('.submission-ids:checked').map(function() {
+                return $(this).val();
+            }).get();
+
+            var form = $('<form>', {
+                method: 'POST',
+                action: $this.data('href'),
+            });
+
+            $.each(submissionIds, function(index, submissionId) {
+                $('<input>', {
+                    type: 'hidden',
+                    name: 'submission_ids',
+                    value: submissionId
+                }).appendTo(form);
+            });
+
+            form.appendTo('body').submit();
+            form.remove();
+        });
+
+        function _disableButtons() {
+            if ($('li.submission-row').length == 0) {
+                $('#export-submissions, #delete-submissions').prop('disabled', true);
+            }
+        }
+
+        $('.js-delete-submission').on('indico:confirmed', function(evt) {
+            evt.preventDefault();
+            var $this = $(this);
+
+            $.ajax({
+                url: $this.data('href'),
+                method: $this.data('method'),
+                data: {
+                    submission_ids: [$this.data('submission-id')]
+                },
+                traditional: true,
+                complete: IndicoUI.Dialogs.Util.progress(),
+                error: handleAjaxError,
+                success: function(data) {
+                    $this.closest('li.submission-row').remove();
+                    _disableButtons();
+                }
+            });
+
+        });
+
+        $('#delete-submissions').on('indico:confirmed', function(evt) {
+            evt.preventDefault();
+            var $this = $(this);
+            var submissionIds = $('.submission-ids:checked').map(function() {
+                return $(this).val();
+            }).get();
+
+            $.ajax({
+                url: $this.data('href'),
+                method: $this.data('method'),
+                data: {
+                    submission_ids: submissionIds
+                },
+                traditional: true,
+                complete: IndicoUI.Dialogs.Util.progress(),
+                error: handleAjaxError,
+                success: function(data) {
+                    $('.submission-ids:checked').parent().remove();
```

Comment:
`.closest('.submission-row')` - that way you aren't tied to a specific dom structure (you might need to add such a class to the row)